lunatik.h

```diff
@@ -138,6 +138,15 @@ static inline void *lunatik_checknull(lua_State *L, void *ptr)
 
 #define lunatik_checkalloc(L, s)	(lunatik_checknull((L), lunatik_malloc((L), (s))))
 
+static inline void lunatik_checkfield(lua_State *L, int idx, const char *field, int type)
+{
+	int tname;
+	lua_getfield(L, idx, field);
+	if ((tname = lua_type(L, -1)) != type)
+		luaL_error(L, "bad field '%s' (%s expected, got %s)", field,
+			lua_typename(L, type), lua_typename(L, tname));
+}
+
```

Comment:
perhaps, though I'm not very found of, it's better to use [GNU extensions](https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html) here.. it might cleanup LUAXTABLE_NEWHOOK a bit..

```suggestion
#define lunatik_checkfield(L, idx, field, type, TYPE)                                           \
do {                                                                                            \
        int _type = lua_getfield(L, idx, field);                                                \
        if (_type != LUA_T#TYPE) {                                                              \
                const char *err = "bad field '%s' (%s expected, got %s)";                       \
                luaL_error(L, err, field, lua_typename(L, ##type), lua_typename(L, _type));       \
        }                                                                                       \
        lua_to#type(L, -1);                                                                     \
} while(0);

```

so we can do..
```
hook->foo = lunatik_checkfield(L, 1, "foo", number, NUMBER);
hook->bar = lunatik_checkfield(L, 1, "bar", string, STRING);
```

what do you think?

Reply:
Initially I had this idea too, but it won't accomodate `LUA_TFUNCTION` in the later PRs.

Reply:
> Is this still needed after the macro addition in the other PR ?

I don't think so.. I think the changes there are good enough regarding this.. we can get merge as is and get back to it later if we think it will be a improvement..